fix: remove pop-up colors from 'cisternas'

LeilaFarias opened the pull request:
This PR removes the popup colors from 'cisternas' visualization to avoid confusion with the legend information.

### How to test it:

1. Go to **'/map?name=cisterna&year=general'**
2. Hover over the states.
  
### The new popup should look like this:

![image](https://github.com/user-attachments/assets/16f03812-5417-46ad-b203-91532ed9d79e)

LeilaFarias committed:
fix: remove pop-up colors from 'cisternas'

LazaroQueiroz reviewed (approved):
nice work!!

RodrigoEC commented on src/components/MapPopup/MapPopup.tsx:
```diff
@@ -44,13 +46,25 @@ const MapPopup: React.FC<MapPopupProps> = ({
-            <TotalArea>{areaInfo.area}</TotalArea>
-          </LineInfo>
-        ))}
+        {colors.length > 0 ? (
+          areas.map((areaInfo: AreaInfo, index) => (
+            <LineInfo key={index}>
+              <Color color={areaInfo.color} $percent={areaInfo.percent || 45} />
+              {areaInfo.percent && (
+                <PercentArea>{areaInfo.percent}%</PercentArea>
+              )}
+              <TotalArea>{areaInfo.area}</TotalArea>
+            </LineInfo>
+          ))
+        ) : (
+          <List>
+            {areas.map((areaInfo, index) => (
+              <ListItem key={index}>
+                <strong>{areaInfo.area}</strong>
+              </ListItem>
+            ))}
+          </List>
+        )}
```
What do you think of moving the outer if statement to the line 52 alone? I think would make the code more readable. The list style could be added by giving the LineInfo component a prop that tell it should appear.

LeilaFarias replied:
i'm not sure i understand it, do you mean like this?: 
```suggestion
{areas.map((areaInfo: AreaInfo, index) => (
          <LineInfo key={index} isColored={colors.length > 0}>
            {colors.length > 0 ? (
              <>
                <Color color={areaInfo.color} $percent={areaInfo.percent || 45} />
                {areaInfo.percent && (
                  <PercentArea>{areaInfo.percent}%</PercentArea>
                )}
                <TotalArea>{areaInfo.area}</TotalArea>
              </>
            ) : (
              <ListItem>
                <strong>{areaInfo.area}</strong>
              </ListItem>
            )}
          </LineInfo>
        ))}
```

RodrigoEC replied:
Not really, I think we could wrap everything with a <List\> and every <LineInfo\> could be a <\i\> tag and if the colors.length if bigger than 0 we could add the <Color> tag and if the areaInfo exists we could add the PercentArea

RodrigoEC commented on src/components/MapPopup/MapPopup.styles.tsx:
```diff
@@ -64,3 +64,15 @@ export const Color = styled.div<{ color: string; $percent: number }>`
 `;
+
+export const List = styled.ul`
+  width: 95%;
```
I personally don't  like when we give such specific values to width - like 95%. I know that this was to solve this issue, right?

![image](https://github.com/user-attachments/assets/4979328a-067b-4c02-8530-343d9b4224ac)

But to solve that we could add a 

```suggestion
   padding-right: 0.5rem;
   box-sizing: border-box;
```
It's a better way to solve this, it ultimately makes kind of the same thing but it's easier to debug in the future.

RodrigoEC commented on src/components/MapPopup/MapPopup.styles.tsx:
```diff
@@ -64,3 +64,15 @@ export const Color = styled.div<{ color: string; $percent: number }>`
+export const ListItem = styled.li`
+  text-align: left;
+  list-style: square;
+  line-height: 1.6;
+  padding: 0.05rem;
+  font-size: 0.9rem;
```
I would remove this 'cause the first to properties don't make a huge difference and the last one

```suggestion
   font-size: 0.8rem;
```

RodrigoEC commented on src/components/MapTiff/MapTiff.tsx:
```diff
@@ -205,9 +205,17 @@ const MapTiff = ({
-              year
-            ]?.imageParams.map((param: any) => param.color);
+
+            // const rasterColors = rasterMetadata.fields.imageData[
+            //   year
+            // ]?.imageParams.map((param: any) => param.color);
+
+            const rasterColors =
+              name === "cisterna"
+                ? []
+                : rasterMetadata.fields.imageData[year]?.imageParams.map(
+                    (param) => param.color,
+                  );
```
Actually liked this solution 'cause it reminds us that this should not be a permanent fix to this "cisterna" problem.

LeilaFarias replied:
Actually, that was the intention lol

RodrigoEC reviewed (changes requested):
Commented some possible refactoring for the code :3

LeilaFarias committed:
fix: adds the new popup layout and fixes  an overflow bug

LeilaFarias committed:
Merge branch 'main' into fix/remove-colors-popup-cisternas

LeilaFarias committed:
fix: fixing bug on rasterColors

LeilaFarias commented:
Hey team, i merged this branch with main and added the requested changes, also fixed this bug of the new layout that's in beta right now

![image](https://github.com/user-attachments/assets/58bc6a2f-d9d3-4c98-88ff-3df1c63c9945)

RodrigoEC reviewed (approved):
Great Work Leila!

![abbot-2](https://github.com/user-attachments/assets/6567a4f3-2295-4844-b1e7-7434c063938a)

cilasmarques commented on src/components/MapTiff/MapTiff.tsx:
```diff
@@ -197,9 +197,16 @@ const MapTiff = ({ isReduced = false, ...props }: { isReduced?: boolean }) => {
-              year
-            ]?.imageParams.map((param: any) => param.color);
+
+            // const rasterColors = rasterMetadata.fields.imageData[
+            //   year
+            // ]?.imageParams.map((param: any) => param.color);
+            const rasterColors =
+              id === "cisterna"
+                ? []
+                : rasterMetadata.fields.imageData[year]?.imageParams.map(
+                    (param) => param.color,
+                  );
```
Refact this in the future
